crates/bevy_ecs/src/system/mod.rs

```diff
@@ -810,6 +810,31 @@ mod tests {
         run_system(&mut world, sys);
     }
 
+    #[test]
+    fn data_set_system() {
+        fn sys(mut _query: Query<DataSet<(&mut A, &A)>>) {}
+        let mut world = World::default();
+        run_system(&mut world, sys);
+    }
+
+    #[test]
+    #[should_panic]
+    fn conflicting_query_with_data_set_system() {
+        fn sys(_query_1: Query<&mut A>, _query_2: Query<DataSet<(&mut A, &B)>>) {}
+
+        let mut world = World::default();
+        run_system(&mut world, sys);
+    }
+
+    #[test]
+    #[should_panic]
+    fn conflicting_data_sets_system() {
+        fn sys(_query_1: Query<DataSet<(&mut A,)>>, _query_2: Query<DataSet<(&mut A, &B)>>) {}
+
+        let mut world = World::default();
+        run_system(&mut world, sys);
+    }
+
```

Comment:
Could you add a should_panic test with the conflict in the second item of the DataSet?